tests/integration/test_record.py

```diff
@@ -347,3 +399,27 @@ def test_records_can_be_merged(app, records_not_merged_in_marcxml):
     with app.test_client() as client:
         assert client.get('/api/literature/111').status_code == 200
         assert client.get('/api/literature/222').status_code == 301
+
+
+def test_records_propagate_links_when_merged(app, records_to_be_merged):
+    before_record_update.disconnect(check_if_record_is_going_to_be_deleted)
+
+    with app.app_context():
+        with db.session.begin_nested():
+            record_to_be_merged = get_db_record('lit', 222)
+            record_to_be_merged['deleted'] = True
+            record_to_be_merged['new_record'] = {
+                '$ref': 'http://localhost:5000/api/literature/111'}
+            record_to_be_merged.commit()
+        db.session.commit()
+
+    with app.test_client() as client:
+        merged_record = json.loads(client.get('/api/literature/222/db').data)
```

Comment:
I think we are doing this awkward thing because `get_db_record` would fail, since the record is now deleted. But actually, as you made me realize, we should rather be testing that record 333's `$ref` is updated, rather than this one!